Refactor backward compatibility tests to use override_qengines decorator

durumu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38838 Refactor backward compatibility tests to use override_qengines decorator**

Differential Revision: [D21676032](https://our.internmc.facebook.com/intern/diff/D21676032)

durumu committed:
Refactor backward compatibility tests to use override_qengines decorator

[ghstack-poisoned]

supriyar commented on test/quantization/test_backward_compatibility.py:
```diff
@@ -11,13 +11,14 @@
 from torch.testing._internal.common_utils import TestCase
-from torch.testing._internal.common_quantized import override_quantized_engine, supported_qengines
+from torch.testing._internal.common_quantized import override_quantized_engine, supported_qengines, override_qengines
+
```
Please fix the lint errors before submitting

jerryzh168 replied:
I think we should also move everything in common_quantized to common_quantization and remove that file. But this change should be a separate PR.

jerryzh168 commented on test/quantization/test_backward_compatibility.py:
```diff
@@ -68,55 +69,50 @@ def remove_prefix(text, prefix):
-                    self._test_op(module_float16, "float16", input_size=[1, 3], input_quantized=False, generate=False)
+        module_qint8 = nnqd.Linear(3, 1, bias_=True, dtype=torch.qint8)
+        self._test_op(module_qint8, "qint8", input_size=[1, 3], input_quantized=False, generate=False)
+        if torch.backends.quantized.engine == 'fbgemm':
```
we can have a function for this check as well

jerryzh168 replied:
please put it in torch/testing/_internal/common_quantization.py

durumu committed:
Update on "Refactor backward compatibility tests to use override_qengines decorator"


Differential Revision: [D21676032](https://our.internmc.facebook.com/intern/diff/D21676032)

[ghstack-poisoned]

durumu committed:
Update on "Refactor backward compatibility tests to use override_qengines decorator"


Differential Revision: [D21676032](https://our.internmc.facebook.com/intern/diff/D21676032)

[ghstack-poisoned]

durumu committed:
Update on "Refactor backward compatibility tests to use override_qengines decorator"


Differential Revision: [D21676032](https://our.internmc.facebook.com/intern/diff/D21676032)

[ghstack-poisoned]

durumu committed:
Update on "Refactor backward compatibility tests to use override_qengines decorator"


Differential Revision: [D21676032](https://our.internmc.facebook.com/intern/diff/D21676032)

[ghstack-poisoned]

durumu committed:
Update on "Refactor backward compatibility tests to use override_qengines decorator"


Differential Revision: [D21676032](https://our.internmc.facebook.com/intern/diff/D21676032)

[ghstack-poisoned]

jerryzh168 reviewed (approved):
Looks good, thanks!

durumu committed:
Update on "Refactor backward compatibility tests to use override_qengines decorator"


Differential Revision: [D21676032](https://our.internmc.facebook.com/intern/diff/D21676032)

[ghstack-poisoned]
